src/Microsoft.ML.Experimental/DataLoadSave/Database/DatabaseLoaderCursor.cs

```diff
@@ -236,79 +236,79 @@ private Delegate CreateGetterDelegate<TValue>(int col)
             private ValueGetter<bool> CreateBooleanGetterDelegate(ColInfo colInfo)
             {
                 int columnIndex = GetColumnIndex(colInfo);
-                return (ref bool value) => value = DataReader.GetBoolean(columnIndex);
+                return (ref bool value) => value = DataReader.IsDBNull(columnIndex) ? default : DataReader.GetBoolean(columnIndex);
             }
 
             private ValueGetter<byte> CreateByteGetterDelegate(ColInfo colInfo)
             {
                 int columnIndex = GetColumnIndex(colInfo);
-                return (ref byte value) => value = DataReader.GetByte(columnIndex);
+                return (ref byte value) => value = DataReader.IsDBNull(columnIndex) ? default : DataReader.GetByte(columnIndex);
             }
 
             private ValueGetter<DateTime> CreateDateTimeGetterDelegate(ColInfo colInfo)
             {
                 int columnIndex = GetColumnIndex(colInfo);
-                return (ref DateTime value) => value = DataReader.GetDateTime(columnIndex);
+                return (ref DateTime value) => value = DataReader.IsDBNull(columnIndex) ? default : DataReader.GetDateTime(columnIndex);
             }
 
             private ValueGetter<double> CreateDoubleGetterDelegate(ColInfo colInfo)
             {
                 int columnIndex = GetColumnIndex(colInfo);
-                return (ref double value) => value = DataReader.GetDouble(columnIndex);
+                return (ref double value) => value = DataReader.IsDBNull(columnIndex) ? default : DataReader.GetDouble(columnIndex);
```

Comment:
I think `null` values for `float` and `double` should get turned into `NaN` instead of `0.0`.